[ChangruHenryQian] iP

ChangruHenryQian opened the pull request:
# Jarvis Chat Bot
> Time for hard work

Jarvis frees your mind of having to remember things you need to do. It's,

+ text-based
+ easy to learn
+ ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 👍 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
    public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Finish Level 0. Rename, Greet, Exit.

Commit:
Finish Level-1: Echo

Commit:
Finish Level 2. Add, List

Commit:
Finish Level 3. Mark as Done

Commit:
Finish Level 4. ToDos, Events, Deadlines

Commit:
Finish A-TextUiTesting

Commit:
Finish Level 5. Handle Errors

Commit:
Finish Level 6. Delete

Commit:
Finish Level 7. Save

Commit:
Update data/duke.txt

Commit:
Merge branch 'branch-Level-7'

rionshocker commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,172 @@
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui = new Ui();
+    private static final String filePath = "./data/duke.txt";
```
Since it is a constant, may I suggest this change?
```suggestion
    private static final String FILE_PATH = "./data/duke.txt";
```

rionshocker reviewed (commented):
Overall well done! Just some minor changes to be made. Can consider abstracting certain methods to different classes.

Commit:
Finish Level 8. Dates and Times

Commit:
Merge branch 'branch-Level-8'

Commit:
Finish A-MoreOOP: Use More OOP

Commit:
Fix bug on delete command

Commit:
Finish A-Packages: Organize into Packages

Commit:
Merge branch 'add-gradle-support'

billieboy7 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    public void parse() {
+        try{
```
using switch statements may help reduce nesting. what do you think?

billieboy7 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,57 @@
+                this.taskList.addTask(newToDo);
+            } else if (command.startsWith("deadline")) {
+                LocalDate deadline = LocalDate.parse(command.split(" /by ", 2)[1]);
+                String name = command.split(" /by ", 2)[0].split(" ", 2)[1];
```
Perhaps splitting them into multiple statements could improve the readability of the code. It can make it easier for you to debug your code too :)

billieboy7 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,83 @@
+package duke;
+
+import java.io.File;
```
I like how you have organised your imports according to their functionality. It's clear and easy to read. Great job!

billieboy7 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,34 @@
+        return "[" + this.getStatusIcon() + "] " + this.description;
+    }
+
+    public String toTxt() {
```
just a trivial issue but would spelling out the txt improve the readability of your code?

Commit:
Finish A-Gradle: Use Gradle

Commit:
Finish A-JUnit: Add JUnit Tests

Commit:
Finish A-Jar: Create a JAR File

Commit:
Add part of Java Doc.

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'master' into branch-A-CodingStandard

Commit:
Implement coding standard.

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Update JavaDoc

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Update JavaDoc

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Start Level 9. Find

Commit:
Merge branch 'master' into branch-Level-9

Commit:
Implement Level 9. Find

Commit:
Merge branch 'branch-Level-9'

Commit:
Start Level 10. GUI

Commit:
Implement Level 10. GUI

Commit:
Merge branch 'branch-Level-10'

Commit:
Update build.gradle

After implementing GUI, the entry point was changed.

Let's change mainClass into duke.Launcher.

Commit:
Implement A-Assertions

The functional code is not capable of checking assumptions.

Assumptions may not be true in some cases and checking is required.

Let's add assert feature to document important assumptions.

Commit:
Improve code quality

Some parts of the code have bad code quality.

Improvements on code quality increase readability.

Let's improve code quality through methods such as refactoring.

ChangruHenryQian committed:
Merge pull request #1 from ChangruHenryQian/branch-A-CodeQuality

Improve code quality

Commit:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/duke/Storage.java

ChangruHenryQian committed:
Merge pull request #2 from ChangruHenryQian/branch-A-Assertions

Implement assertions

Commit:
Implement B-FixedDurationTasks

Commit:
Add welcome message into GUI

Commit:
Add a representative screenshot

Commit:
Implement A-UserGuide: User Guide

Commit:
Update user guide

Commit:
Implement CheckStyle

Commit:
Add test cases

Commit:
Update test cases

Commit:
Update test cases

Commit:
Improve JavaDoc

Commit:
Update README.md

Commit:
Improve JavaDoc

Commit:
Update user guide

Commit:
Add test cases

Commit:
Update JavaDoc
